### ISSUE TITLE
Outlets array expects kebab-cased names instead of camel-cased ones (like targets do)

### ISSUE THREAD

**Issue by wynksaiddestroy** [hotwired/stimulus#615]
If you have a **target** name which consists of multiple words, you'll define your targets as follows:
```
static targets = [ "errorMessage" ]
```
An **outlet** name with multiple words, has to be defined the following way:
```
static outlets = [ "error-message" ]
``` 
If this is the intended behaviour, it would be nice to add a small notice to the documentation. Thanks!

**Comment by marcoroth** [comment]
Hey @wynksaiddestroy, thanks for opening this issue!

This is a good point, and I think I agree with you that it's kinda unexpected if you look at how targets are defined.

I'm wondering why I did it that way. I guess because controller identifiers are also being written in kebab-case, and since outlets reference controller identifiers I probably went with that.

I guess you could expect both ways, depending on how you look at it.

Not sure how we should proceed here. I'd be fine with changing it. But that would probably also mean it's a breaking change.

**Comment by marcoroth** [comment]
Good point! I'd be happy to change it, but would also love to hear feedback from others.

**Comment by connorwbey** [comment]
Agree it should be changed for consistency. Might as well be now while the feature is new.

**Comment by dhh** [comment]
Would be happy to ship a quick 3.2.2 that addresses this. If there was a way to detect kebab case, still have it work with a warning of depreciation, and then document and recommend camel going forward, that would be great.

**Comment by minimul** [comment]
I like kabab-case and wish targets worked as outlets do it that regard – but yeah consistency overrules.

**Comment by drjayvee** [comment]
This may be unrelated, but the docs leave me wondering whether you need to use the outlet controller name verbatim, or whether you're allowed to alias it.

```
<div class="stuff" data-controller="awesome-stuff"></div>

<div
  data-controller="search"
  data-search-awesome-stuff-outlet=".stuff"
>

<!-- is the following allowed as well -->
<div
  data-controller="search"
  data-search-stuff-outlet=".stuff"
>
```

Would the 2nd then be allowed to use
```js
class extends Controller {
  static outlets = ['stuff']
  declare readonly stuffOutlet: AwesomeStuffController
}
```

**Comment by rupasix** [comment]
> Originally, outlet getter had this problem.
> However, it used to be discernible at the time of definition and it was predictable that the two would be mixed.

Concerning is, if you use controllers in directory, your outlet controller names in camelcase are simplified, eg: `search--form` and `search-form` will be `this.searchFormOutlet`.

**Comment by minimul** [comment]
@drjayvee 

I tested this and it seems like the outlet name needs to be verbatim (to the controller you want "outlet" to) and an alias doesn't work. 

Is that what you found?

**Comment by drjayvee** [comment]
> @drjayvee 
> 
> I tested this and it seems like the outlet name needs to be verbatim (to the controller you want "outlet" to) and an alias doesn't work. 
> 
> Is that what you found?

@minimul yes, that's what I found as well.

**Comment by adiakritos** [comment]
I want to add to this that if you want to reference an outlet whose file name is something like:

`foo/bar-controller` 

Then the html kebab case is needed when setting the outlet inside the controller like so:

static outlets = [ "foo--bar"].

Whether it's changed to match the pattern used by targets or not, I think the docs should speak to this since it's the current way newcomers will expect things to be given the pattern established by targets.
